mautic/mautic
# Sugarcrm7 integration fullsync

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? N| 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Update the SugarCRM plugin to enable full synchronization of Mautic Leads to SugarCRM version 7. 
All functionnalities are similar to SugarCRM 6.x plugin [](https://github.com/mautic/mautic/pull/4247)
Developped by @canal-web and @webmecanik. 

#### Steps to test this PR:
1. Create new Oauth keys in a SugarCRM 7.X instance
Log in as admin, click on admin menu, chose Oauth link. Then choose "create Oauth key" on Oauth top menu. OAuth Version for the key should be 2.0.
2. On the Mautic side, configure the SugarCRM plugin (Sugar URL, keys, login/password)
3. On the Mautic side, configure the plugin functionnalities
4. On the Mautic side, configure the fields mapping for contacts/leads and companies
5. Run php app/console mautic:integration:fetchleads --integration=Sugarcrm
or php app/console mautic:integration:fetchleads --fetch-all --integration=Sugarcrm to synchronize all leads.
6. Optionnaly install, on the SugarCRM V7 side, the webActivity plugin, and launch
php app/console mautic:integration:pushleadactivity
The zip from the webActivity plugin should be download from :
https://drive.google.com/open?id=0B7f6B2OD0DBrbGRISTJNNkRVVzQ


#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
